models/repo_list.go

```diff
@@ -97,103 +97,124 @@ type SearchRepoOptions struct {
 	// Owner in we search search
 	//
 	// in: query
-	OwnerID     int64  `json:"uid"`
-	Searcher    *User  `json:"-"` //ID of the person who's seeking
-	OrderBy     string `json:"-"`
-	Private     bool   `json:"-"` // Include private repositories in results
-	Collaborate bool   `json:"-"` // Include collaborative repositories
-	Starred     bool   `json:"-"`
-	Page        int    `json:"-"`
-	IsProfile   bool   `json:"-"`
+	OwnerID     int64         `json:"uid"`
```

Comment:
As far as I understand, the meaning of the `OwnerID` field depends on the value of the `Starred` field. If `Starred` is false, we look for repos owned by `OwnerID` (plus possibly collaborative repos). If `Starred` is true, however, we look for repos starred by `OwnerID`.

I think it would be more straightforward to have the `OwnerID` always have the first meaning (look for repos owned by `OwnerID`, plus possibly collaborative repos). The `Starred` field can then be replaced with a `StarredByID` field, indicating that we should look for repos that are starred by `StarredByID` (if `StarredByID` is 0, then don't filter by stars).

Reply:
@ethantkoenig Yes for first paragraph (as I mentioned in https://github.com/go-gitea/gitea/pull/2371#issuecomment-324942147).

I partialy agree with second part. I don't like to mix 2 IDs in one function, how can I know what part of result is because of one or another ID (same problem is now with `OwnerID` and `Searcher`)? I think change is needed. For example if you set `starred` flag, it should mean "include starred (by owner) repositories", second option can be like filter saying "give me only starred repositories" (same as "give me only forks/mirrors/source/..." like in #2326).

Anyway, this change should be done in another PR to keep commits small.